### PR TITLE
Move recipient_interceptor to :staging gem group

### PR DESCRIPTION
**Pull request by delphaber**
Since recipient interceptor is used only in staging, could we move the gem in the appropriate group?

### CODE REVIEW

File: templates/Gemfile.erb [hunk 1]
```diff
@@ -56,3 +55,7 @@ end
 group :staging, :production do
   gem "rack-timeout"
 end
+
+group :staging do
```
Comment (edwardloveall) [inline templates/Gemfile.erb]: I think it should also be in development and test. You don't want to be sending emails there either.

**Comment by tute** [comment]
Thanks! I'm ok with having it globally. That way we can register interceptors in environments without needing to think if code is loaded or not, the library does nothing if no interceptors are registered (like in production), and the library is tiny.

**Comment by edwardloveall** [comment]
Even if it's tiny, I still feel like it's unnecessary to have it in production at all. Even tiny code can cause problems, and if you can avoid those problems, I think it's better.

**Comment by mike-burns** [comment]
I do agree with that reasoning, but I also worry about staging and production being different. If even tiny code can cause problems, I want to catch those on staging first.

**Comment by edwardloveall** [comment]
That's a good reason I hadn't considered. It's awkward to have a gem you're not intending to use, but you're right, bugs are worse.

**Comment by tute** [comment]
Leaving it globally, thanks for your work!

**Pull request closed without merging**